src/server/http/xsrf.js

```diff
@@ -4,18 +4,48 @@ export default function (kbnServer, server, config) {
   const disabled = config.get('server.xsrf.disableProtection');
   const versionHeader = 'kbn-version';
   const xsrfHeader = 'kbn-xsrf';
+  const contentTypeHeader = 'content-type';
+  const allowedRequestMediaTypes = ['application/json', 'application/x-ndjson'];
 
   server.ext('onPostAuth', function (req, reply) {
     if (disabled) {
       return reply.continue();
     }
 
     const isSafeMethod = req.method === 'get' || req.method === 'head';
-    const hasVersionHeader = versionHeader in req.headers;
-    const hasXsrfHeader = xsrfHeader in req.headers;
+    if (!isSafeMethod) {
+      const hasVersionHeader = versionHeader in req.headers;
+      const hasXsrfHeader = xsrfHeader in req.headers;
 
-    if (!isSafeMethod && !hasVersionHeader && !hasXsrfHeader) {
-      return reply(badRequest(`Request must contain an ${xsrfHeader} header`));
+      if (hasXsrfHeader) {
+        server.log(
+          ['warning', 'deprecation'],
+          `The ${xsrfHeader} header is deprecated and will be removed in a future version of Kibana.` +
+          ` Specify a ${contentTypeHeader} header of either application/json or application/x-ndjson instead.`
```

Comment:
> 1. I think we should explore testing the server.logs. We can maybe sinon.stub the server.log method and ensure it's called correctly. Would be great to check the callCount on the stub too.

I believe it should be easy to do with `sinon`. I'll look into it. At this stage I'm trying to get a feeling of the "test-verbosity level" agreement you have on the team. Usually I didn't test such fine details, but since it's visible and used by the user why not!

> 2. When this is logged you can already have a correct content-type header specified, so we shouldn't include the second line, as it might get people confused.

Maybe we can avoid additional `if/ternary` clause by just rephrasing the second sentence a bit? Something like `CSRF protection requires only a ${contentTypeHeader} header of either application/json or application/x-ndjson.`?

> hm, ideally as this is related to security we should link people to the docs where our handling of xsrf is explain?

Good idea, do you have an example in the codebase how links to the docs are usually provided (e.g. with proper Kibana version tag and so on)?